packages/gatsby-plugin-google-analytics/src/gatsby-node.js

```diff
@@ -1,7 +1,39 @@
-exports.onPreInit = ({ reporter }, options) => {
-  if (!options.trackingId) {
-    reporter.warn(
-      `The Google Analytics plugin requires a tracking ID. Did you mean to add it?`
-    )
-  }
-}
```

Comment:
@mxstbr Shouldn't we have kept `onPreInit` in case `GATSBY_EXPERIMENTAL_PLUGIN_OPTION_VALIDATION` wasn't `true`? 🤔

If people update `gatsby-plugin-google-analytics` but don't update `gatsby`, they won't get any warning anymore.

Reply:
True, good catch! PR welcome 👍

Reply:
On it boss! 😉

Reply:
Here you go: https://github.com/gatsbyjs/gatsby/pull/27495 🙂